lib/gtfs/data.ex

```diff
@@ -274,43 +280,52 @@ defmodule Gtfs.Data do
     )
   end
 
-  @spec timepoint_ids_for_routes([RoutePattern.t()], MapSet.t(Route.id()), trips_by_id()) ::
-          timepoint_ids_by_route()
-  defp timepoint_ids_for_routes(route_patterns, route_ids, trips) do
+  @spec all_timepoints_by_id(binary()) :: timepoints_by_id()
+  defp all_timepoints_by_id(timepoints_data) do
+    timepoints_data
+    |> Csv.parse(parse: &Timepoint.from_csv_row/1)
+    |> Enum.reduce(%{}, fn timepoint, acc ->
+      Map.put(acc, timepoint.id, timepoint)
+    end)
```

Comment:
Instead of a raw `reduce`, this could be `|> Map.new(fn timepoint -> {timepoint.id, timepoint} end)`